src/generic_factory.h

```diff
@@ -103,23 +104,83 @@ const my_class &string_id<my_class>::obj() const
 
 \endcode
 */
+
+template<typename T>
+class string_id_reader;
+
 template<typename T>
 class generic_factory
 {
     protected:
-        std::unordered_map< string_id<T>, T> data;
+        std::vector<T> list;
+        std::unordered_map<string_id<T>, int_id<T>> map;
 
         std::string type_name;
         std::string id_member_name;
+        std::string alias_member_name;
 
         T &load_override( const string_id<T> &id, JsonObject &jo ) {
             T obj;
+
             obj.id = id;
             obj.load( jo );
             obj.was_loaded = true;
-            T &result = data[id];
-            result = std::move( obj );
-            return result;
+
+            T &inserted_obj = insert( obj );
+
+            if( !alias_member_name.empty() && jo.has_member( alias_member_name ) ) {
+                const int_id<T> i_id = map[id];
+                std::vector<string_id<T>> aliases;
+
+                mandatory( jo, obj.was_loaded, alias_member_name, aliases, string_id_reader<T> {} );
+
+                for( const auto &alias : aliases ) {
+                    const auto iter = map.find( alias );
+
+                    if( iter != map.end() ) {
```

Comment:
You can remove the iterator and just call `map::count` to find out if the key is contained.